src/Tizen.NUI.Scene3D/src/public/ModelMotion/MotionData.cs

```diff
@@ -16,109 +16,384 @@
+                ret = new MotionIndex(cPtr, true);
+            }
+            else
             {
-                return motionValues[(int)index].Item1;
+                // We found matched NUI animation. Reduce cPtr reference count.
+                HandleRef handle = new HandleRef(this, cPtr);
+                Interop.MotionIndex.DeleteMotionIndex(handle);
+                handle = new HandleRef(null, IntPtr.Zero);
             }
-            return null;
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
         }
 
         /// <summary>
         /// Get MotionValue at index'th. null if invalid index inputed
         /// </summary>
+        /// <param name="index">The index of motion data list</param>
+        /// <returns>The index'th MotionValue. Or empty handle that doesn't have body</returns>
         [EditorBrowsable(EditorBrowsableState.Never)]
         public MotionValue GetValue(uint index)
         {
-            if(motionValues != null && index < motionValues.Count)
+            IntPtr cPtr = Interop.MotionData.GetValue(SwigCPtr, index);
+            MotionValue ret = Registry.GetManagedBaseHandleFromNativePtr(cPtr) as MotionValue;
+            if (ret == null)
+            {
+                // Register new animation into Registry.
+                ret = new MotionValue(cPtr, true);
+            }
+            else
             {
-                return motionValues[(int)index].Item2;
+                // We found matched NUI animation. Reduce cPtr reference count.
+                HandleRef handle = new HandleRef(this, cPtr);
+                Interop.MotionValue.DeleteMotionValue(handle);
+                handle = new HandleRef(null, IntPtr.Zero);
             }
-            return null;
+            if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
+            return ret;
         }
 
         /// <summary>
-        /// Clear all inputed values.
+        /// Load bvh animation.
+        /// Scale is additional scale factor of bvh animation. It is possible that
+        /// Model's scale may not matched with bvh animation scale.
+        /// If scale is null, default use as Vector3.ONE
+        /// After load completed, <see cref="LoadCompleted"/> event will be invoked.
         /// </summary>
+        /// <param name="bvhFilename">Name of bvh format file.</param>
+        /// <param name="scale">Scale value of bvh animation match with model.</param>
+        /// <param name="synchronousLoad">Load synchronously or not. Default is async load.</param>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public void Clear()
+        public void LoadBvhAnimation(string bvhFilename, Vector3 scale = null, bool synchronousLoad = false)
```

Comment:
This naming is quite good because Bvh is very common format.
But bvh is one of format type for motion capture data.
So, when we support other motion capture format in any future than we need to make new method for the format.
So how about to use LoadMotionCaptureAnimation as name?
Then we can reuse the method when we support other formats.

Reply:
Your point looks good to me! `LoadBvh` was too file specific API so it will be hard ti apend more jobs ib future. I make API name as `LoadMotionCaptureAnimation' niw.